graylog2-web-interface/src/components/users/TimeoutInput.tsx

```diff
@@ -92,7 +94,8 @@ const TimeoutInput = ({ value: propsValue, onChange }: Props) => {
                help="When checked, sessions never time out due to inactivity."
                formGroupClassName="no-bm"
                onChange={_onClick}
-               checked={sessionTimeoutNever} />
+               checked={sessionTimeoutNever}
+               disabled={isGlobalTimeoutEnabled} />
```

Comment:
I think we should use the `isGlobalTimeoutEnabled` as a condition to display the help message or hide the timeout fields instead of disabling them.